build/configs/rtl8721csm/rtl8721csm_download.sh

```diff
@@ -275,19 +280,79 @@ download_all()
 		echo "Downloading ${parts[$partidx]} binary"
 		echo "=========================="
 
-		./amebad_image_tool $TTYDEV 1 ${offsets[$partidx]} ${exe_name}
-		rm ${exe_name}
-
+		./amebad_image_tool "download" $TTYDEV 1 ${offsets[$partidx]} ${exe_name}
 	done
 	echo ""
 	echo "Download COMPLETE!"
 }
 
+erase()
+{
+	cd ${IMG_TOOL_PATH}
+	echo "Starting Erase..."
+	ota_addr=$(($FLASH_START_ADDR + $CONFIG_AMEBAD_FLASH_CAPACITY))
+	if [[ $2 == "all" || $2 == "ALL" ]];then
+		found_kernel=false
+		flash_ota=false
+		echo ""
+		echo "=========================="
+		echo "      Erasing All"
+		echo "=========================="
+		./amebad_image_tool "erase" $TTYDEV 1 $FLASH_START_ADDR 0 $(($CONFIG_AMEBAD_FLASH_CAPACITY>>10))
+	else
+		for partidx in ${!parts[@]}; do
+			if [[ $2 == "kernel" || $2 == "KERNEL" ]];then
+				if [[ "${parts[$partidx]}" != "kernel" ]];then
+					continue
+				elif [[ ${offsets[$partidx]} -lt ${ota_addr} ]];then
+					ota_addr=0
+					found_kernel=false
+				else
+					continue
+				fi
+			elif [[ $2 == "ota" || $2 == "OTA" ]];then
+				if [[ "${parts[$partidx]}" != "kernel" ]];then
+					continue
+				elif [[ ${offsets[$partidx]} -lt ${ota_addr} ]];then
+					ota_addr=${offsets[$partidx]}
+					continue
+				else
+					flash_ota=false
+				fi
+			elif [[ $2 == "userfs" || $2 == "USERFS" ]];then
+				if [[ "${parts[$partidx]}" != "userfs" ]];then
+					continue
+				fi
+			fi
```

Comment:
I think `else` is needed for unexpected arguments like `./dbuild download erase`, `./dbuild download erase abcd`

Reply:
Hi @jeongarmy , it is done in the most recent commit.